src/bodies.cpp

```diff
@@ -759,9 +781,9 @@ void bodies::ConvexMesh::useDimensions(const shapes::Shape *shape)
   mesh_data_->bounding_cylinder_.radius = maxdist;
   mesh_data_->bounding_cylinder_.length = cyl_length;
 
-  FILE* null = fopen ("/dev/null","w");
+  static FILE* null = fopen ("/dev/null","w");
```

Comment:
It is somewhat of an ad-hoc fix, but definitely better than before.

We can move the FILE* to file scope and add a mutex to construct the null device. On the other hand: Isn't the worst case that we leak a small number of FILE*?